java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        if (elseStatement != null) {
+          // if statement will create the required block.
+          if (!elseStatement.is(Tree.Kind.IF_STATEMENT)) {
+            currentBlock = createBlock(next);
+          }
+          build(elseStatement);
+          elseBlock = currentBlock;
+        }
+        // process then-branch
+        currentBlock = createBlock(next);
+        build(ifStatementTree.thenStatement());
+        Block thenBlock = currentBlock;
+        // process condition
+        currentBlock = createBranch(ifStatementTree, thenBlock, elseBlock);
+        buildCondition(ifStatementTree.condition(), thenBlock, elseBlock);
+        break;
+      }
+      case CONDITIONAL_EXPRESSION: {
+        ConditionalExpressionTree cond = (ConditionalExpressionTree) tree;
+        Block next = currentBlock;
+        // process else-branch
+        ExpressionTree elseStatement = cond.falseExpression();
+        currentBlock = createBlock(next);
+        build(elseStatement);
+        Block elseBlock = currentBlock;
+        // process then-branch
+        currentBlock = createBlock(next);
+        build(cond.trueExpression());
+        Block thenBlock = currentBlock;
+        // process condition
+        currentBlock = createBranch(cond, thenBlock, elseBlock);
+        buildCondition(cond.condition(), thenBlock, elseBlock);
+        break;
+      }
+      case VARIABLE:
+        currentBlock.elements.add(tree);
+        VariableTree variableTree = (VariableTree) tree;
+        if (variableTree.initializer() != null) {
+          build(variableTree.initializer());
+        }
+        break;
+      case MULTIPLY:
+      case DIVIDE:
+      case REMAINDER:
+      case PLUS:
+      case MINUS:
+      case LEFT_SHIFT:
+      case RIGHT_SHIFT:
+      case UNSIGNED_RIGHT_SHIFT:
+      case AND:
+      case XOR:
+      case OR:
+      case GREATER_THAN:
+      case GREATER_THAN_OR_EQUAL_TO:
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL_TO:
+      case EQUAL_TO:
+      case NOT_EQUAL_TO:
+        BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(binaryExpressionTree.rightOperand());
+        build(binaryExpressionTree.leftOperand());
+        break;
+      case ASSIGNMENT:
+      case LEFT_SHIFT_ASSIGNMENT:
+      case RIGHT_SHIFT_ASSIGNMENT:
+      case AND_ASSIGNMENT:
+      case REMAINDER_ASSIGNMENT:
+      case UNSIGNED_RIGHT_SHIFT_ASSIGNMENT:
+      case OR_ASSIGNMENT:
+      case XOR_ASSIGNMENT:
+      case DIVIDE_ASSIGNMENT:
+      case MULTIPLY_ASSIGNMENT:
+      case PLUS_ASSIGNMENT:
+      case MINUS_ASSIGNMENT:
+        AssignmentExpressionTree assignmentExpressionTree = (AssignmentExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(assignmentExpressionTree.variable());
+        build(assignmentExpressionTree.expression());
+        break;
+      case MEMBER_SELECT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)